### ISSUE TITLE
Users are not able to specify the WebProxy property on the XXX

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus.Transport.AzureServiceBus#531]
The configuration API doesn't currently expose a way to set the [`WebProxy`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusclientoptions.webproxy?view=azure-dotnet#azure-messaging-servicebus-servicebusclientoptions-webproxy) property on the [`ServiceBusClientOptions`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusclientoptions) and thus preventing a proxy to be used.

This has been reported on [StackOverflow](https://stackoverflow.com/q/71921976)

A potential solution would be to expose a way for users to pass in an instance of the ServiceBusClientOptions that we would use as is

**Comment by SeanFeldman** [comment]
@andreasohlund passing a constructed client could be an option as well. Assuming Core can read the settings out of the passed object for its own configuration. That way, the client can be configured and by the user code and handed over to the transport to use as is.

**Comment by SeanFeldman** [comment]
By accepting a client, NSB would more friendly to the scenarios where customization of the underlying SDK is required. Today it’s impossible. With a client passed in, it will no longer be a constraint.

**Comment by mauroservienti** [comment]
Yep, I agree. We should be doing the same thing we do in SQS. By default, the transport builds a client, but users can pass a pre-configured client. That approach greatly simplifies all client customizations and even future features the client SDK might introduce.

**Comment by SeanFeldman** [comment]
Clients injected for ASQ transport, Azure persistence, data bus. Time to align this transport. And out that `candidate-for-next-release` label.

**Comment by danielmarbach** [comment]
Agreed

**Comment by andreasohlund** [comment]
I think this would be an easy change, since we need both the regular client and the administrative client (to set up subscriptions etc) we would need to add a ctor that requires both to be passed? Is there a better way? (I couldn't find a way to derive the admin client from the regular client)

Ie:

```
var serviceBusClient = new ServiceBusClient(myConnectionString);
var serviceBusAdministrationClient = new ServiceBusAdministrationClient(myConnectionString);

var asbTransport = new AzureServiceBusTransport(serviceBusClient, serviceBusAdministrationClient);

endpointConfiguration.UseTranport(asbTransport);
```

**Comment by danielmarbach** [comment]
In Azure Table Persistence, Cosmos and also ASQ we made it possible to register providers to that you can fully control the clients via DI. I think that is the best model. Example

https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/master/src/Transport/Config/ICloudTableClientProvider.cs#L9

**Comment by andreasohlund** [comment]
So both a ctor https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/master/src/Transport/AzureStorageQueueTransport.cs#L73 and a provider?

**Comment by danielmarbach** [comment]
That's what we did in ASQ and the persistences yes

**Comment by danielmarbach** [comment]
Given that the ability to provide a ServiceBusClient is far from trivial as [outlined in the previous comment, ](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/issues/531#issuecomment-1270267353)we have added a web proxy property in this PR https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/659
